frontend/vue/components/UserAccount/UserAccountLayout.vue

```diff
@@ -0,0 +1,196 @@
+  <div class="user-account">
+    <nav class="user-account__section-nav">
+      <div class="user-account__section-nav__user-data">
+        <span class="user-account__section-nav__user-data__name">{{ userName }}</span>
+        <span class="user-account__section-nav__user-data__role">{{ userRole }}</span>
+      </div>
+      <div class="user-account__section-nav__link-list">
+        <AppLink
+          v-for="{displayName, hash} in sectionList"
+          :key="hash"
+          class="user-account__section-nav__link-list__link"
+          :class="{'user-account__section-nav__link-list__link_active': activeSection === hash}"
+          :url="hash"
+          target="_self"
+        >
+          {{ displayName }}
+        </AppLink>
+        <AppLink
+          class="user-account__section-nav__link-list__link"
+          url="#"
+          target="_self"
+        >
+          {{ $translate('Log Out') }}
+        </AppLink>
+      </div>
+    </nav>
+    <bx-dropdown
+      class="user-account__section-dropdown"
+      :value="activeSection"
+      @bx-dropdown-selected="switchPanel($event)"
+    >
+      <bx-dropdown-item
+        v-for="{displayName, hash} in sectionList"
+        :key="hash"
+        class="user-account__section-dropdown__item"
+        :value="hash"
+      >
+        {{ displayName }}
+      </bx-dropdown-item>
+    </bx-dropdown>
+    <section class="user-account__section-container">
+      <div v-if="activeSection === sectionList[0].hash">
+        PROGRESS SECTION
+      </div>
+      <div v-if="activeSection === sectionList[1].hash">
+        GROUPS SECTION
+      </div>
+      <div v-if="activeSection === sectionList[2].hash">
+        BADGES SECTION
+      </div>
+      <div v-if="activeSection === sectionList[3].hash">
+        PRIVACY SECTION
+      </div>
+    </section>
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent } from 'vue-demi'
+import AppLink from '../common/AppLink.vue'
+
+export default defineComponent({
+  name: 'UserAccountLayout',
+  components: {
+    AppLink
+  },
+  props: {
+    userName: { type: String, required: false, default: 'Unknown' },
+    userRole: { type: String, required: false, default: 'Unknown' }
+  },
+  data () {
+    return {
+      activeSection: '',
+      sectionList: [
+        {
+          displayName: this.$translate('My Learning'),
+          hash: '#MyLearning'
+        },
+        {
+          displayName: this.$translate('Groups'),
+          hash: '#Groups'
+        },
+        {
+          displayName: this.$translate('Badges'),
+          hash: '#Badges'
+        },
+        {
+          displayName: this.$translate('Privacy'),
+          hash: '#Privacy'
+        }
+      ]
+    }
+  },
+  mounted () {
+    this.activeSection = window.location.hash || this.sectionList[0].hash
+    window?.addEventListener('hashchange', () => {
+      this.activeSection = window.location.hash
+    }, false)
+  },
+  methods: {
+    switchPanel (event: any) {
```

Comment:
This might be typed as:

```suggestion
    switchPanel (event: CustomEvent) {
```